BuildingBlocks/src/BuildingBlocks.API/Serilog/BackboneIdMaskingOperator.cs

```diff
@@ -0,0 +1,12 @@
+﻿using Serilog.Enrichers.Sensitive;
+
+namespace Backbone.BuildingBlocks.API.Serilog;
+
+public class BackboneIdMaskingOperator : RegexMaskingOperator
+{
+    public BackboneIdMaskingOperator(string initials, int length = -1) : base(IsStringOfConstantLength(length) ? RegexPattern(initials, length) : RegexPattern(initials)) { }
```

Comment:
This is kind of ugly. The parameter is called "initials", but it can contain the initials or a RegEx. And depending on a parameter named "length" you can control what "initials" is used as.

I'd suggest factory methods `ForRegex` and `ForId`.

Reply:
I created the factory methods taking into account the chat we had on the topic.

Reply:
Now that we only have one usecase for this, we might as well use the constructor you had before. ;)

Reply:
Hah, no prob, done :)